File: 1217-relative-sort-array/1217-relative-sort-array.java
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        int[] freq = new int[1001];
        for(int n : arr1) freq[n]++;
        int i = 0;
        for(int n : arr2) while(freq[n]-- > 0) arr1[i++] = n;
        for(int j = 0; j<1001; j++) while(freq[j]-- > 0) arr1[i++] = j;
        return arr1;
    }
}

File: 1217-relative-sort-array/1217-relative-sort-array.js
/**
 * @param {number[]} arr1
 * @param {number[]} arr2
 * @return {number[]}
 */
var relativeSortArray = function(arr1, arr2) {
    const lookup = new Map();
    const N = arr2.length;
    arr2.forEach((a, i) => {
        lookup.set(a, i);
    });
    return arr1.sort((a, b) => {
        a = lookup.has(a) ? lookup.get(a) : N + a;
        b = lookup.has(b) ? lookup.get(b) : N + b;
        return a - b;
    });
};